src/python/pants/build_graph/bundle_mixin.py

```diff
@@ -0,0 +1,95 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import os.path
+
+from pants.base.build_environment import get_buildroot
+from pants.base.exceptions import TargetDefinitionException
+from pants.build_graph.app_base import AppBase
+from pants.fs import archive
+from pants.util.dirutil import absolute_symlink, safe_mkdir, safe_mkdir_for
+from pants.util.fileutil import atomic_copy
+
+
+class BundleMixin(object):
+
+  @staticmethod
+  def register_common_options(register):
```

Comment:
Will update with a comment - this seems better than copy/paste of the options to ensure all bundles have consistent options. This seems like an opportunity to refactor into `BundleBase` in the future, especially to simplify adding bundles for more languages, or a language-neutral bundle target.

Reply:
I think the question is, why is this a separately-named method, and not an override of the regular `register_options`. Since all `register_options` implementations must call `super(...).register_options`, subclasses can always add their own options too.

Reply:
Updated so `BundleMixin` now subclasses `TaskBase` like other mixin's do, so now setting options works like you would expect. In `PythonBundle` we no longer even have a `register_options` method because we get it from the mixin, and in `BundleCreate` we get the common options by calling super, and add a custom option.